src/assets/assetMetrics.spec.js

```diff
@@ -0,0 +1,476 @@
+      let assetMetricFromServerAfterFormat;
+      let assetMetricFromServerBeforeFormat;
+      let assetMetricToServerAfterFormat;
+      let assetMetricToServerBeforeFormat;
+      let assetTypeId;
+      let promise;
+      let request;
+      let toCamelCase;
+      let toSnakeCase;
+
+      beforeEach(function() {
+        assetMetricFromServerAfterFormat = fixture.build('assetMetric');
+        assetMetricFromServerBeforeFormat = fixture.build('assetMetric', null, {
+          fromServer: true
+        });
+        assetMetricToServerAfterFormat = fixture.build('assetMetric', null, {
+          fromServer: true
+        });
+        assetMetricToServerBeforeFormat = fixture.build('assetMetric');
+
+        assetTypeId = fixture.build('assetType').id;
+
+        request = {
+          ...baseRequest,
+          post: this.sandbox.stub().resolves(assetMetricFromServerBeforeFormat)
+        };
+        toCamelCase = this.sandbox
+          .stub(objectUtils, 'toCamelCase')
+          .returns(assetMetricFromServerAfterFormat);
+        toSnakeCase = this.sandbox
+          .stub(objectUtils, 'toSnakeCase')
+          .returns(assetMetricToServerAfterFormat);
+
+        const assetMetrics = new AssetMetrics(baseSdk, request, expectedHost);
+
+        promise = assetMetrics.create(
+          assetTypeId,
+          assetMetricToServerBeforeFormat
+        );
+      });
+
+      it('formats the submitted asset metric object to send to the server', function() {
+        expect(toSnakeCase).to.be.deep.calledWith(
+          assetMetricToServerBeforeFormat
+        );
+      });
+
+      it('creates a new asset metric', function() {
+        expect(request.post).to.be.deep.calledWith(
+          `${expectedHost}/assets/types/${assetTypeId}/metrics`,
+          assetMetricToServerAfterFormat
+        );
+      });
+
+      it('formats the returned asset metric object', function() {
+        return promise.then(() => {
+          expect(toCamelCase).to.be.deep.calledWith(
+            assetMetricFromServerBeforeFormat
+          );
+        });
+      });
+
+      it('returns a fulfilled promise with the new asset metric information', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.deep.equal(
+          assetMetricFromServerAfterFormat
+        );
+      });
+    });
+
+    context('when there is missing required information', function() {
+      let assetMetric;
+      let assetMetrics;
+      let assetTypeId;
+      let promise;
+
+      beforeEach(function() {
+        assetMetric = fixture.build('assetMetric');
+        assetMetrics = new AssetMetrics(baseSdk, baseRequest, expectedHost);
+        assetTypeId = fixture.build('assetType').id;
+      });
+
+      it('throws an error if there is no asset type ID provided', function() {
+        promise = assetMetrics.create(null, assetMetric);
+
+        return expect(promise).to.be.rejectedWith(
+          'An asset type ID is required for creating an asset metric.'
```

Comment:
Can we switch this to `to create` so it matches the other error message?

Reply:
👍 Yeah, that sounds great